File: src/app/schemas/marketplace.py
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel
from typing import List, Optional


class PublicacionBase(BaseModel):
    modelo: Optional[str] = None
    capacidad_gb: Optional[int] = None
    color: Optional[str] = None
    imei: Optional[str] = None
    bateria_porcentaje: Optional[int] = None
    estado_estetico: Optional[str] = None
    estado_funcional: Optional[str] = None
    titulo: Optional[str] = None
    descripcion: Optional[str] = None
    precio_publicado: Optional[Decimal] = None
    estado: Optional[str] = None
    fotos_urls: Optional[List[str]] = None


class PublicacionCreate(PublicacionBase):
    """Si enviás `Authorization: Bearer` (access), `id_usuario` se toma del token.
    Si no hay token, debe indicarse `id_usuario` en el cuerpo (p. ej. panel admin)."""

    id_usuario: Optional[int] = None
    fecha_publicacion: Optional[datetime] = None


class PublicacionUpdate(BaseModel):
    modelo: Optional[str] = None
    capacidad_gb: Optional[int] = None
    color: Optional[str] = None
    imei: Optional[str] = None
    bateria_porcentaje: Optional[int] = None
    estado_estetico: Optional[str] = None
    estado_funcional: Optional[str] = None
    titulo: Optional[str] = None
    descripcion: Optional[str] = None
    precio_publicado: Optional[Decimal] = None
    estado: Optional[str] = None
    fecha_publicacion: Optional[datetime] = None
    fotos_urls: Optional[List[str]] = None


class PublicacionResponse(PublicacionBase):
    id_publicacion: int
    id_usuario: int
    fecha_publicacion: Optional[datetime] = None

    class Config:
        from_attributes = True


class RevisionPublicacionBase(BaseModel):
    estado_revision: Optional[str] = None
    observaciones: Optional[str] = None


class RevisionPublicacionCreate(RevisionPublicacionBase):
    id_publicacion: int
    fecha_revision: Optional[datetime] = None


class RevisionPublicacionUpdate(BaseModel):
    estado_revision: Optional[str] = None
    observaciones: Optional[str] = None
    fecha_revision: Optional[datetime] = None


class RevisionPublicacionResponse(RevisionPublicacionBase):
    id_revision: int
    id_publicacion: int
    fecha_revision: Optional[datetime] = None

    class Config:
        from_attributes = True


class InteresPublicacionBase(BaseModel):
    mensaje: Optional[str] = None
    estado: Optional[str] = None


class InteresPublicacionCreate(InteresPublicacionBase):
    id_publicacion: int


class InteresPublicacionUpdate(BaseModel):
    mensaje: Optional[str] = None
    estado: Optional[str] = None


class InteresPublicacionResponse(InteresPublicacionBase):
    id_interes: int
    id_publicacion: int
    id_usuario_interesado: int
    fecha_interes: Optional[datetime] = None
    comprador_nombre: Optional[str] = None
    comprador_email: Optional[str] = None
    comprador_telefono: Optional[str] = None
    publicacion_titulo: Optional[str] = None
    publicacion_modelo: Optional[str] = None
    whatsapp_url: Optional[str] = None

    class Config:
        from_attributes = True

File: client/src/types/marketplace.ts
/** Alineado con `app/schemas/marketplace.py` */

export type Publicacion = {
  id_publicacion: number
  id_usuario: number
  modelo: string | null
  capacidad_gb: number | null
  color: string | null
  imei: string | null
  bateria_porcentaje: number | null
  estado_estetico: string | null
  estado_funcional: string | null
  titulo: string | null
  descripcion: string | null
  precio_publicado: string | number | null
  estado: string | null
  fecha_publicacion: string | null
  fotos_urls: string[] | null
  tiene_caja: boolean | null
  tiene_cargador: boolean | null
  vendedor_nombre: string | null
  vendedor_telefono: string | null
}

export type RevisionPublicacion = {
  id_revision: number
  id_publicacion: number
  estado_revision: string | null
  observaciones: string | null
  fecha_revision: string | null
}

export type InteresPublicacion = {
  id_interes: number
  id_publicacion: number
  id_usuario_interesado: number
  mensaje: string | null
  estado: string | null
  fecha_interes: string | null
  comprador_nombre: string | null
  comprador_email: string | null
  comprador_telefono: string | null
  publicacion_titulo: string | null
  publicacion_modelo: string | null
  whatsapp_url: string | null
}
